sdk/containerinstance/mgmt/src/main/java/com/azure/resourcemanager/containerinstance/implementation/ContainerGroupImpl.java

```diff
@@ -0,0 +1,811 @@
+                            .flatMap(
+                                profile -> {
+                                    creatableNetworkProfileName = null;
+                                    creatableNetworkProfileInner = null;
+                                    return Mono.empty();
+                                }));
+        }
+        return mono;
+    }
+
+    @Override
+    protected Mono<ContainerGroupInner> createInner() {
+        this.containerGroupMsiHandler.processCreatedExternalIdentities();
+        this.containerGroupMsiHandler.handleExternalIdentities();
+        final ContainerGroupImpl self = this;
+
+        if (!isInCreateMode()) {
+            Resource resource = new Resource();
+            resource.withLocation(self.regionName());
+            resource.withTags(self.tags());
+            return beforeCreation()
+                .then(
+                    manager()
+                        .inner()
+                        .getContainerGroups()
+                        .updateAsync(self.resourceGroupName(), self.name(), resource));
+        } else if (newFileShares == null || creatableStorageAccountKey == null) {
+            return beforeCreation()
+                .then(manager().inner().getContainerGroups().createOrUpdateAsync(resourceGroupName(), name(), inner()));
+        } else {
+            final StorageAccount storageAccount = this.taskResult(this.creatableStorageAccountKey);
+            return beforeCreation()
+                .thenMany(createFileShareAsync(storageAccount))
+                .map(
+                    volumeParameters ->
+                        this
+                            .defineVolume(volumeParameters.volumeName)
+                            .withExistingReadWriteAzureFileShare(volumeParameters.fileShareName)
+                            .withStorageAccountName(storageAccount.name())
+                            .withStorageAccountKey(volumeParameters.storageAccountKey)
+                            .attach())
+                .then(
+                    this
+                        .manager()
+                        .inner()
+                        .getContainerGroups()
+                        .createOrUpdateAsync(resourceGroupName(), name(), inner()));
+        }
+    }
+
+    private static class VolumeParameters {
+        private String volumeName;
+        private String fileShareName;
+        private String storageAccountKey;
+
+        VolumeParameters(String volumeName, String fileShareName, String storageAccountKey) {
+            this.volumeName = volumeName;
+            this.fileShareName = fileShareName;
+            this.storageAccountKey = storageAccountKey;
+        }
+    }
+
+    private Flux<VolumeParameters> createFileShareAsync(final StorageAccount storageAccount) {
+        return storageAccount
+            .getKeysAsync()
+            .map(storageAccountKeys -> storageAccountKeys.get(0).value())
```

Comment:
remember to run `credcheck` for storage key

Reply:
All playback tests are not using the `Share File Volume`, so no keys are recorded.